src/streaming-client.ts

```diff
@@ -0,0 +1,97 @@
+import * as fs from 'fs';
+import * as sockets from 'websocket';
+import * as events from 'events';
+import { Duplex, PassThrough } from 'stream';
+
+import AudioConfig from './models/streaming/AudioConfig';
+import BufferedDuplex from './models/streaming/BufferedDuplex';
+import StreamingResponse from './models/streaming/StreamingResponse';
+
+export class RevAiStreamingClient extends events.EventEmitter {
+    accessToken: string;
+    config: AudioConfig;
+    baseUrl: string;
+    requests: PassThrough;
+    responses: PassThrough;
+    client: sockets.WebSocketClient;
+    constructor(accessToken: string, config: AudioConfig, version = 'v1alpha'){
+        super();
+        this.accessToken = accessToken;
+        this.config = config;
+        this.baseUrl = `wss://api.rev.ai/speechtotext/${version}/stream?` +
+            `access_token=${accessToken}` +
+            `&content_type=${config.getContentTypeString()}`;
+        this.requests = new PassThrough();
+        this.responses = new PassThrough({ objectMode: true });
+        this.client = new sockets.client();
+    }
+
+    start (): Duplex {
+        var self = this;
+        this.client.on('connectFailed', function(error: Error) {
+            self.emit('connectFailed', error);
+            self.requests.end();
+        });
+        this.client.on('connect', function(connection: any) {
+            self.emit('connect');
+            connection.on('error', function(error : Error) {
+                self.emit('error', error);
+                self.requests.end();
+                self.responses.push(null);
+            });
+            connection.on('close', function(code: number, reason: string) {
+                self.emit('close', code, reason);
+                self.requests.end();
+                self.responses.push(null);
+            });
+            connection.on('message', function(message: any) {
+                if (message.type === 'utf8') {
+                    self.responses.write(JSON.parse(message.utf8Data) as StreamingResponse);
+                }
+            });
+            
+            function sendFromBuffer() {
```

Comment:
personal preference, but you could make this not a local function, by having `sendFromBuffer(connection: any)` and pass the connection through

Reply:
thoughts on this comment?

Reply:
I prefer to keep it within the connection. I view it as a function which is part of the connection not a part of the client as a whole.